src/main/java/io/cryostat/recordings/JvmIdHelper.java

```diff
@@ -135,6 +135,10 @@ public ServiceRef resolveId(ServiceRef sr) throws JvmIdGetException {
         }
         URI serviceUri = sr.getServiceUri();
         String uriStr = serviceUri.toString();
+        if (sr.getJvmId() != null) {
+            reverse.put(sr.getJvmId(), sr);
```

Comment:
From what I see, this seems like this line will happen in all code branches. Put it at the beginning of the method?